pkg/controller/storagecluster/reconcile.go

```diff
@@ -176,12 +201,23 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		// to set readiness.
 		r := ready.NewFileReady()
 		err = r.Set()
-		instance.Status.Phase = statusutil.PhaseReady
 		if err != nil {
 			reqLogger.Error(err, "Failed to mark operator ready")
 			return reconcile.Result{}, err
 		}
+		if conditionsv1.IsStatusConditionTrue(instance.Status.Conditions, conditionsv1.ConditionAvailable) &&
+			conditionsv1.IsStatusConditionTrue(instance.Status.Conditions, conditionsv1.ConditionUpgradeable) &&
+			conditionsv1.IsStatusConditionFalse(instance.Status.Conditions, conditionsv1.ConditionProgressing) &&
+			conditionsv1.IsStatusConditionFalse(instance.Status.Conditions, conditionsv1.ConditionDegraded) {
+			instance.Status.Phase = statusutil.PhaseReady
+		} else {
+			// waiting for Ceph Cluster to be Ready
+			instance.Status.Phase = statusutil.PhaseNotReady
+		}
 	} else {
+		if instance.Status.Phase != statusutil.PhaseClusterExpanding {
```

Comment:
I don't think you can assume PhaseError just because it's not PhaseClusterExpanding. It could be NotReady or Progressing. You can determine that by looking at status.conditions.

Reply:
ack.